lms/djangoapps/ccx/api/v0/tests/test_views.py

```diff
@@ -175,15 +198,24 @@ def test_authorization(self):
         for auth in auth_list:
             resp = self.client.get(self.list_url_master_course, {}, HTTP_AUTHORIZATION=auth)
             self.assertEqual(resp.status_code, status.HTTP_401_UNAUTHORIZED)
+
         resp = self.client.get(self.list_url_master_course, {}, HTTP_AUTHORIZATION=self.auth)
         self.assertEqual(resp.status_code, status.HTTP_200_OK)
 
+        # test for oauth2_provider
+        resp = self.client.get(self.list_url_master_course, {}, HTTP_AUTHORIZATION=self.auth_header_oauth2_provider)
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
```

Comment:
Repeating each test case for the other authorization string like this is resulting in a lot of code duplication.  Could you instead use [ddt](http://ddt.readthedocs.io/en/latest/example.html) to parameterize the relevant tests to be automatically repeated with each option?  Something like this:

```python
AUTH_ATTRS = ('auth', 'auth_header_oauth2_provider')

...

    @ddt.data(*AUTH_ATTRS)
    def test_authorization(self, auth_attr):
        ...
        resp = self.client.get(self.list_url_master_course, {}, HTTP_AUTHORIZATION=getattr(self, auth_attr))
        ...
```

This would result in a little runtime repetition of setup work (building URLs, etc.) but should make it significantly easier to read and maintain the tests.